lightning/src/ln/channelmanager.rs

```diff
@@ -1499,15 +1481,18 @@ macro_rules! handle_chan_restoration_locked {
 				// only case where we can get a new ChannelMonitorUpdate would be if we also
 				// have some commitment updates to send as well.
 				assert!($commitment_update.is_some());
-				if let Err(e) = $self.chain_monitor.update_channel($channel_entry.get().get_funding_txo().unwrap(), monitor_update) {
-					// channel_reestablish doesn't guarantee the order it returns is sensical
-					// for the messages it returns, but if we're setting what messages to
-					// re-transmit on monitor update success, we need to make sure it is sane.
-					let mut order = $order;
-					if $raa.is_none() {
-						order = RAACommitmentOrder::CommitmentFirst;
+				match $self.chain_monitor.update_channel($channel_entry.get().get_funding_txo().unwrap(), monitor_update) {
+					ChannelMonitorUpdateResult::UpdateComplete => {},
+					e => {
```

Comment:
Would be more readable if the other two variants were explicitly used here. And would force consideration if any new variants are added. Same in a few other places.